fairlearn/metrics/_plotter.py

```diff
@@ -0,0 +1,297 @@
+        The label corresponding to the confidence interval bars
+    """
+    color = ax.lines[0].get_color() if kind == "scatter" else "black"
+
+    # extend legend with user-provided legend_label
+    handles, labels = ax.get_legend_handles_labels()
+    custom_line = [Line2D([0], [0], color=color, label=legend_label)]
+    handles.extend(custom_line)
+    ax.legend(handles=handles)
+
+
+def _plot_df(
+    df, metrics, kind, subplots, legend_label, df_all_errors=None, **kwargs
+):
+    r"""Plot the data with or without errors.
+
+    Parameters
+    ----------
+    df : pd.DataFrame
+        The collection of disaggregated metric values, along with the metric errors.
+
+    metrics : str or list of str
+        The name of the metrics to plot.
+        Should match columns from the given :class:`fairlearn.metrics.MetricFrame`.
+
+    kind : str
+        The type of plot to display. e.g. "bar", "line", etc.
+        List of options is detailed in :meth:`pandas.DataFrame.plot`
+
+    subplots : bool
+        Whether or not to plot metrics on separate subplots
+
+    legend_label : str
+        The label corresponding to the confidence interval bars
+
+    df_all_errors : pd.DataFrame, default=None
+        Optional argument for DataFrame that indicates the errors to plot
+
+    \*\*kwargs
+        Keyword arguments that are passed in to :meth:`pandas.DataFrame.plot`.
+
+    Returns
+    -------
+    :class:`matplotlib.axes.Axes` or :class:`numpy.ndarray` of them
+    """
+    if df_all_errors is not None:
+        yerr = np.array(
+            [
+                np.array([[*row] for row in df_all_errors[metric]]).T
+                for metric in metrics
+            ]
+        )
+        if kind == "scatter":
+            axs = df[metrics].plot(
+                linestyle="",
+                marker="o",
+                yerr=yerr,
+                subplots=subplots,
+                **kwargs
+            )
+        else:
+            axs = df[metrics].plot(
+                kind=kind, yerr=yerr, subplots=subplots, **kwargs
+            )
+
+        if isinstance(axs, np.ndarray):
+            for ax in axs.flatten():
+                _build_legend(ax, kind, legend_label)
+        else:
+            _build_legend(axs, kind, legend_label)
+
+    else:
+        if kind == "scatter":
+            axs = df[metrics].plot(
+                linestyle="", marker="o", subplots=subplots, **kwargs
+            )
+        else:
+            axs = df[metrics].plot(kind=kind, subplots=subplots, **kwargs)
+
+    return axs
+
+
+def plot_metric_frame(
+    metric_frame: MetricFrame,
+    *,
+    kind: str = "scatter",
+    metrics: Union[List[str], str] = None,
+    conf_intervals: Union[List[str], str] = None,
+    subplots: bool = True,
+    plot_error_labels: bool = False,
+    error_labels_precision: int = 4,
+    error_labels_fontsize: int = 8,
+    error_labels_color: str = "black",
+    error_labels_ha: str = "center",
+    legend_label: str = "95% CI",
```

Comment:
I think we should change the argument name from `legend_label` to `error_labels_legend` (for consistency with other arguments). Also, the default should be `None` corresponding to no legend being displayed. We shouldn't assume that the provided confidence intervals are 95%.

As an alternative to the default of `None`, we could also default to `"conf. intervals"`, I think that `"CI"` is way too cryptic. I'm personally slightly in favor of `None`, but I don't mind `"conf. interval"` since that matches the name of the provided argument. (Tagging @hildeweerts since she mentioned this before.)